Document clusters/{id}/metric_queries/alerts endpoint

cben opened the pull request:
(https://issues.redhat.com/browse/SDA-1919, part 1, more to come in separate PRs)

modeled on #102.  For some reason it's not getting into generated openapi for me — the types are added but there is still no `metric_queries/alerts` endpoint :man_shrugging:

jhernand commented on model/clusters_mgmt/v1/alert_info_type.model:
```diff
@@ -0,0 +1,24 @@
+
+// Provides information about a single alert firing on the cluster.
+struct AlertInfo {
+	// The alert name.  Multiple alerts with same name are possible (and the API does not currently expose details to distinguish them).
```
Use shorter lines; 100 characters max is good in my opinion.

jhernand commented on model/clusters_mgmt/v1/alert_info_type.model:
```diff
@@ -0,0 +1,24 @@
+	Name     string
+
+	// The alert severity.
+        Severity AlertSeverity
```
There seems to be a mix of spaces and tabs indenting this file. Make sure to use only tabs.

jhernand commented on model/clusters_mgmt/v1/alerts_info_type.model:
```diff
@@ -0,0 +1,20 @@
+limitations under the License.
+*/
+
+// Provides detailed information about the operators installed on the cluster.
```
This comment isn't accurate.

jhernand commented:
There used to be an issue with generation of the OpenAPI. It was fixed in version 0.0.23 of the metamodel. See https://github.com/openshift-online/ocm-api-metamodel/pull/71. It was quite a subtle Go behaviour.

With the current version of the model (0.0.25) it works correctly and generates the following:

```json
    "/api/clusters_mgmt/v1/clusters/{cluster_id}/metric_queries/alerts": {
      "get": {
        "parameters": [
          {
            "name": "cluster_id",
            "in": "path",
            "schema": {
              "type": "string"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "Success.\n",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/AlertsInfo"
                }
              }
            }
          },
          "default": {
            "description": "Error.\n",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Error"
                }
              }
            }
          }
        }
      }
    },
```

In https://github.com/openshift-online/ocm-api-model/pull/115 I am adding a new `openapi` target to the _Makefile_ so that you can easily generate the _OpenAPI_ specification and check it.

cben commented:
Right but I'm running freshly checked out metamodel, I see that fix in the code, and adding debugging prints I see Paths() does not return the alerts path.  investigating...

cben commented:
Aaah, OK, I _was_ using an old version because Makefile used to checkout `metamodel_version:=v0.0.25` :man_facepalming:.  New version does include new path.

Addressed feedback, PTAL
